### PR TITLE
Fix crash when damaging off-screen hint

### PR DESCRIPTION
**Pull request by kchibisov**

### CODE REVIEW

File: alacritty/src/display/mod.rs [hunk 1]
```diff
@@ -1061,10 +1061,15 @@ impl Display {
...
-            for point in (hint.bounds.start().line.0..=hint.bounds.end().line.0).flat_map(|line| {
-                term::point_to_viewport(display_offset, Point::new(Line(line), Column(0)))
-            }) {
+            for mut point in
+                (hint.bounds.start().line.0..=hint.bounds.end().line.0).flat_map(|line| {
+                    term::point_to_viewport(display_offset, Point::new(Line(line), Column(0)))
+                })
+            {
+                // Clamp hint point to viewport bounds, since it could be off-screen.
+                point.line = point.line.clamp(0, last_line);
```
Comment (chrisduerr) [inline alacritty/src/display/mod.rs:1072]: Feels like we should just skip this point, rather than clamping?

**Review by chrisduerr — CHANGES_REQUESTED** [review]

File: alacritty/src/display/mod.rs (hunk 1 above)
Comment (kchibisov) [reply alacritty/src/display/mod.rs:1072]: Hm, makes sense.

File: alacritty/src/display/mod.rs [hunk 2]
```diff
@@ -1061,9 +1061,11 @@ impl Display {
     #[inline]
     fn damage_highlighted_hints<T: EventListener>(&self, terminal: &mut Term<T>) {
         let display_offset = terminal.grid().display_offset();
+        let last_line = terminal.screen_lines() - 1;
```
Comment (chrisduerr) [inline alacritty/src/display/mod.rs:1064]:
```suggestion
        let last_line = terminal.topmost_line();
```

File: alacritty/src/display/mod.rs [hunk 3]
```diff
@@ -1061,9 +1061,11 @@ impl Display {
...
         for hint in self.highlighted_hint.iter().chain(&self.vi_highlighted_hint) {
             for point in (hint.bounds.start().line.0..=hint.bounds.end().line.0).flat_map(|line| {
                 term::point_to_viewport(display_offset, Point::new(Line(line), Column(0)))
+                    .filter(|line| line.line <= last_line)
```
Comment (chrisduerr) [inline alacritty/src/display/mod.rs:1068]:
```suggestion
                    .filter(|point| point.line <= last_line)
```

**Review by chrisduerr — CHANGES_REQUESTED** [review]

File: alacritty/src/display/mod.rs [hunk 4]
```diff
@@ -1061,9 +1061,11 @@ impl Display {
     #[inline]
     fn damage_highlighted_hints<T: EventListener>(&self, terminal: &mut Term<T>) {
         let display_offset = terminal.grid().display_offset();
+        let last_line = terminal.topmost_line();
```
Comment (chrisduerr) [inline alacritty/src/display/mod.rs:1064]:
```suggestion
        let topmost_line = terminal.topmost_line();
```

File: alacritty/src/display/mod.rs [hunk 5]
```diff
@@ -1061,9 +1061,11 @@ impl Display {
...
         for hint in self.highlighted_hint.iter().chain(&self.vi_highlighted_hint) {
             for point in (hint.bounds.start().line.0..=hint.bounds.end().line.0).flat_map(|line| {
                 term::point_to_viewport(display_offset, Point::new(Line(line), Column(0)))
+                    .filter(|point| point.line <= last_line)
```
Comment (chrisduerr) [inline alacritty/src/display/mod.rs:1068]:
```suggestion
                    .filter(|point| point.line <= topmost_line)
```

Just for consistency.

**Review by chrisduerr — CHANGES_REQUESTED** [review]

File: alacritty/src/display/mod.rs (hunk 4 above)
Comment (kchibisov) [reply alacritty/src/display/mod.rs:1064]: Thta's not topmost line, topmost is in the grid history, but I want the viewport one.

File: alacritty/src/display/mod.rs (hunk 2 above)
Comment (chrisduerr) [reply alacritty/src/display/mod.rs:1064]: Oh right…

Comment (chrisduerr) [reply alacritty/src/display/mod.rs:1064]: Maybe we want to name this viewport_top instead? Just to be explicit?

Comment (kchibisov) [reply alacritty/src/display/mod.rs:1064]: Well, we can call it `last_screen_line`?

Comment (chrisduerr) [reply alacritty/src/display/mod.rs]: Decided to go with just `last_visible_line`, I think that should be fine. I don't think there's anything *perfect*, so no point endlessly bikeshedding about variable naming.

**Review by chrisduerr — APPROVED** [review]

**Pull request merged**